src/Library/public/usbguard/IPCServer.cpp

```diff
@@ -281,41 +288,55 @@ namespace usbguard
     const std::string& device_rule,
     uint32_t rule_id)
   {
-    d_pointer->DevicePolicyChanged(id, target_old, target_new, device_rule, rule_id);
+    for (const auto& d_pointer : d_pointers) {
+      d_pointer->DevicePolicyChanged(id, target_old, target_new, device_rule, rule_id);
+    }
   }
 
   void IPCServer::PropertyParameterChanged(const std::string& name,
     const std::string& value_old,
     const std::string& value_new)
   {
-    d_pointer->PropertyParameterChanged(name, value_old, value_new);
+    for (const auto& d_pointer : d_pointers) {
+      d_pointer->PropertyParameterChanged(name, value_old, value_new);
+    }
   }
 
   void IPCServer::ExceptionMessage(const std::string& context,
     const std::string& object,
     const std::string& reason)
   {
-    d_pointer->ExceptionMessage(context, object, reason);
+    for (const auto& d_pointer : d_pointers) {
+      d_pointer->ExceptionMessage(context, object, reason);
+    }
   }
 
   void IPCServer::addAllowedUID(uid_t uid, const IPCServer::AccessControl& ac)
   {
-    d_pointer->addAllowedUID(uid, ac);
+    for (const auto& d_pointer : d_pointers) {
+      d_pointer->addAllowedUID(uid, ac);
```

Comment:
What's the intended behavior if any of the IPC implementation fail? Should we check the return values?

Reply:
There aren't return values to check. Though I did go through and convert some error log statements to exceptions, so the D-Bus IPC server better matches the behavior of the libqb IPC server code.